### PR TITLE
smart assembly scaffold

### PR DESCRIPTION
**Pull request by rhubarbgarden**
- users can build dapps using their own wallet of choice
- add world explorer
- add minimal toggle on/off contract example

Screenshot 2024-12-19 at 10 54 37
Screenshot 2024-12-19 at 10 56 32

### CODE REVIEW

File: smart-assembly-scaffold/packages/contracts/.envsample
Comment (CCP-Red-Dragon) [inline smart-assembly-scaffold/packages/contracts/.envsample:1]: Would you be able to change it to .env if possible?

Comment (rhubarbgarden) [reply smart-assembly-scaffold/packages/contracts/.envsample:1]: This is better left as `.envsample`. It's not best practice to check in `.env` as that frequently changes in local development. `.envsample` provides a template for what each dev's `.env` should look like without checking in their local environment.

File: smart-assembly-scaffold/packages/client/src/mud/useWorldContract.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        },
+      });
+    },
+    staleTime: Infinity,
+    refetchOnMount: false,
+    refetchOnReconnect: false,
+    refetchOnWindowFocus: false,
+  });
+
+  // console.log(client, sessionClient)
```
Comment (CCP-Grogu) [inline smart-assembly-scaffold/packages/client/src/mud/useWorldContract.ts:59]: image

**Review by CCP-Grogu — APPROVED** [review]

**Review by CCP-Red-Dragon — CHANGES_REQUESTED** [review]
There seem to be runtime issues with connecting the wallet in-game + in the web browser.

**Review by CCP-Red-Dragon — APPROVED** [review]

**Pull request merged**